File: lab-1/domain/individual_playlist_manager.py
from content_manager import ContentManager


class IndividualPlaylistManager(ContentManager):
    def __init__(self, name):
        super().__init__(name)
        self.playlist = None  # A single playlist managed

    def set_playlist(self, playlist):
        self.playlist = playlist

    def add_item(self, item):
        if self.playlist:
            self.playlist.add_item(item)

    def remove_item(self, item):
        if self.playlist:
            self.playlist.remove_item(item)

    def display_contents(self):
        if self.playlist:
            self.playlist.display_info()

File: src/main/kotlin/content_manager/implementations/IndividualPlaylistManager.kt
package content_manager.implementations
import content_manager.ContentManager
import media_item.MediaItem
import media_item.implementations.Playlist
import items_manager.implementations.PlaylistItemsManager

class IndividualPlaylistManager(override val name: String) : ContentManager {
    var playlist: Playlist? = null
    var playlistItemsManager: PlaylistItemsManager? = null

    fun setIndividualPlaylist(playlist: Playlist) {
        this.playlist = playlist
        this.playlistItemsManager = PlaylistItemsManager(playlist)
    }

    override fun addItem(item: MediaItem) {
        if (playlist != null) {
            playlistItemsManager?.addItem(item)
        }
    }

    override fun removeItem(item: MediaItem) {
        if (playlist != null) {
            playlistItemsManager?.removeItem(item)
        }
    }

    override fun displayContents() {
        playlist?.displayInfo()
    }
}